core/src/main/java/org/opensearch/sql/expression/function/PrometheusFunctions.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function;
+
+import static org.opensearch.sql.data.type.ExprCoreType.LONG;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+import org.opensearch.sql.expression.env.Environment;
+
+
+/**
+ * Utility Class for prometheus connector specific functions.
+ */
+@UtilityClass
+public class PrometheusFunctions {
+
+  public static final String QUERY = "query";
+  public static final String STARTTIME = "starttime";
+  public static final String ENDTIME = "endtime";
+  public static final String STEP = "step";
+
+  /**
+   * Add table functions specific to Prometheus to repository.
+   */
+  public void register(BuiltinFunctionRepository repository) {
+    repository.register(queryRange());
+  }
+
+  private static FunctionResolver queryRange() {
```

Comment:
It is worth to create a `PrometheusFunctionResolver` which extends `DefaultFunctionResolver` and holds all related logic. It should be placed in a separate file.